source_http.go

```diff
@@ -86,6 +86,15 @@ func (s *HTTPImageSource) setAuthorizationHeader(req *http.Request, ireq *http.R
 	}
 }
 
+func (s *HTTPImageSource) setForwardHeaders(req *http.Request, ireq *http.Request) {
+	headers := s.Config.ForwardHeaders
+	for _, header := range headers {
+		if header != "" && len(ireq.Header[header]) > 0 {
```

Comment:
Here I used this approach to check if a header is defined or if it's empty. Leaving the `header != ""` condition alone brings the scenario where all the forwarded headers are set although the original request hasn't defined them.

Reply:
Oh wow, of course you're completely right! What I meant to say was that we only need to check for existence, rather than that it has a value. So basically:

```suggestion
		if ireq.Header.Has(header) && req.Header.Has(header) {
```

Since you've cleaned up s.Config.ForwardHeaders, I suppose `header != ""` became effectively a noop.

Reply:
Oh great, I didn't know that I could check for existence that way in a map. I will submit the changes.